buildbot/master/files/config/factories.py

```diff
@@ -58,20 +58,23 @@ class DynamicServoFactory(ServoFactory):
 
     def __init__(self, builder_name, environment):
         self.environment = environment
+        self.is_windows = re.match('windows.*', builder_name) is not None
         try:
             config_dir = os.path.dirname(os.path.realpath(__file__))
             yaml_path = os.path.join(config_dir, 'steps.yml')
             with open(yaml_path) as steps_file:
                 builder_steps = yaml.safe_load(steps_file)
             commands = builder_steps[builder_name]
-            dynamic_steps = [self.make_step(command) for command in commands]
+            dynamic_steps = [self.make_step(command, self.is_windows)
+                             for command in commands]
         except Exception as e:  # Bad step configuration, fail build
             print(str(e))
             dynamic_steps = [BadConfigurationStep(e)]
 
-        # TODO: windows compatibility (use a custom script for this?)
-        pkill_step = [steps.ShellCommand(command=["pkill", "-x", "servo"],
-                                         decodeRC={0: SUCCESS, 1: SUCCESS})]
+        pkill_step = [steps.ShellCommand(
+            command=self.make_pkill_command("servo"),
+            decodeRC={0: SUCCESS, 1: SUCCESS}
```

Comment:
Does this `decodeRC` apply for the powershell command as well? I think it would be better if the `make_pkill_command` function returned a `steps.ShellCommand` directly instead of just the command array.